tests/report/common/tests.cpp

```diff
@@ -301,144 +328,294 @@ TEST_FCN void TestLocalReport(void* args_)
+        report_buffer_ptr, report_ptr_size, false, zeros, OE_REPORT_DATA_SIZE);
+    oe_free_report(report_buffer_ptr);
+}
+}
 
-        /* Failure cases. */
-        OE_TEST(
-            oe_get_target_info(NULL, report_size, &target, &target_size) ==
-            OE_INVALID_PARAMETER);
+/* oe_get_target_info scenario.
+ *   a. Extract the target_info from the report and use that
+ *      as the opt_params field.
+ *   b. Ensure oe_get_target_info fails on improper inputs.
+ */
+{
+    sgx_target_info_t target;
+    sgx_target_info_t* target_ptr;
+    size_t target_size = 0;
+    size_t target_ptr_size = 0;
 
-        OE_TEST(
-            oe_get_target_info(
-                report_buffer,
-                sizeof(oe_report_header_t) + sizeof(sgx_report_t) - 1,
-                &target,
-                &target_size) == OE_INVALID_PARAMETER);
+    OE_TEST(
+        GetReport_v1(0, NULL, 0, NULL, 0, report_buffer, &report_size) ==
+        OE_OK);
+    ValidateReport(
+        report_buffer, report_size, false, zeros, OE_REPORT_DATA_SIZE);
 
-        OE_TEST(
-            oe_get_target_info(report_buffer, report_size, &target, NULL) ==
-            OE_INVALID_PARAMETER);
+    OE_TEST(
+        oe_get_target_info_v1(
+            report_buffer, report_size, &target, &target_size) ==
+        OE_BUFFER_TOO_SMALL);
+    OE_TEST(target_size == sizeof(target));
 
-        target_size = sizeof(target) + 1;
-        OE_TEST(
-            oe_get_target_info(
-                report_buffer, report_size, NULL, &target_size) ==
-            OE_BUFFER_TOO_SMALL);
-        OE_TEST(target_size == sizeof(target));
+    OE_TEST(
+        oe_get_target_info_v1(
+            report_buffer, report_size, &target, &target_size) == OE_OK);
+    OE_TEST(target_size == sizeof(target));
 
-        target_size = sizeof(target) - 1;
-        OE_TEST(
-            oe_get_target_info(
-                report_buffer, report_size, &target, &target_size) ==
-            OE_BUFFER_TOO_SMALL);
-        OE_TEST(target_size == sizeof(target));
-    }
+    OE_TEST(
+        oe_get_target_info_v2(
+            report_buffer,
+            report_size,
+            (void**)&target_ptr,
+            &target_ptr_size) == OE_OK);
+    OE_TEST(target_ptr_size == sizeof(target));
+    oe_free_target_info(target_ptr);
+
+    target_ptr = NULL;
+    target_ptr_size = 0;
+    OE_TEST(
+        oe_get_target_info(
+            report_buffer,
+            report_size,
+            (void**)&target_ptr,
+            &target_ptr_size) == OE_OK);
+    OE_TEST(target_ptr_size == sizeof(target));
+    oe_free_target_info(target_ptr);
 
-    /*
-     * OE_SMALL_BUFFER scenarios:
-     *     a. NULL buffer
-     *     b. Size too small.
-     */
-    {
-        report_size = 1024 * 1204;
-        OE_TEST(
-            GetReport(0, NULL, 0, NULL, 0, NULL, &report_size) ==
-            OE_BUFFER_TOO_SMALL);
-        OE_TEST(report_size == OE_LOCAL_REPORT_SIZE);
+    OE_TEST(
+        GetReport_v1(
```

Comment:
The `get_report/validate_report` should be run on `target_ptr` to make sure that they have the correct info in them.